pkg/container-utils/containerd/containerd.go

```diff
@@ -15,31 +15,185 @@
 package containerd
 
 import (
+	"context"
+	"fmt"
 	"time"
 
-	criclient "github.com/inspektor-gadget/inspektor-gadget/pkg/container-utils/cri"
+	"github.com/containerd/containerd"
+	tasks "github.com/containerd/containerd/api/services/tasks/v1"
+	"github.com/containerd/containerd/api/types/task"
+	"google.golang.org/grpc"
+
 	runtimeclient "github.com/inspektor-gadget/inspektor-gadget/pkg/container-utils/runtime-client"
 )
 
 const (
 	DefaultTimeout = 2 * time.Second
 )
 
+const (
+	ContainerNameNerdctl = "nerdctl/name"
+	ContainerNameK8s     = "io.kubernetes.container.name"
+)
+
 type ContainerdClient struct {
-	criclient.CRIClient
+	client *containerd.Client
 }
 
 func NewContainerdClient(socketPath string) (runtimeclient.ContainerRuntimeClient, error) {
 	if socketPath == "" {
 		socketPath = runtimeclient.ContainerdDefaultSocketPath
 	}
 
-	criClient, err := criclient.NewCRIClient(runtimeclient.ContainerdName, socketPath, DefaultTimeout)
+	client, err := containerd.New(socketPath,
+		containerd.WithTimeout(DefaultTimeout),
+		containerd.WithDefaultNamespace("k8s.io"),
+		containerd.WithDialOpts([]grpc.DialOption{
+			grpc.WithBlock(),
```

Comment:
I had to remove this to make it works.

Reply:
Me too, forgot to push the changes though